### PR TITLE
test: add tests for plugin itself

### LINKED ISSUE #18: Test plugin itself
There should be tests that run the plugin and check for the desired output.

### PR DESCRIPTION
**Pull request by lukashass**
Closes #18 

Updates devDependencies, as well.

### CODE REVIEW

File: test/envConfig.test.ts [hunk 1]
```diff
@@ -49,3 +58,109 @@ describe('createEnvConfigContent', () => {
...
+      expect(response.status).toBe(200);
+      expect(response.text).toContain('<script src="/env-config.js"></script>');
+    });
+
+    it('serves /env-config.js', async () => {
+      expect.assertions(2);
+
+      // given
+      const variables = ['BACKEND_URL'];
+      process.env.BACKEND_URL = 'http://localhost:4000';
```
Comment (anbraten) [inline test/envConfig.test.ts:105]:
```suggestion
      const myEnv = { BACKEND_URL: 'http://localhost:4000' };
      const variables = Object.keys(myEnv);
      process.env = { ...process.env, ...myEnv };
```

File: test/envConfig.test.ts [hunk 2]
```diff
@@ -49,3 +58,109 @@ describe('createEnvConfigContent', () => {
...
+
+      // then
+      expect(indexHtml.toString()).toContain('<script src="/env-config.js"></script>');
+    });
+
+    it('creates env-config.template.js containing variables and placeholders', async () => {
+      expect.assertions(1);
+
+      // given
+      const variables = ['BACKEND_URL'];
```
Comment (anbraten) [inline test/envConfig.test.ts:152]: Maybe it would be nice to move this up to not wirte it again

**Review by anbraten — APPROVED** [review]

File: test/envConfig.test.ts (hunk 2 above)
Comment (lukashass) [reply test/envConfig.test.ts:152]: There are tests with other `variables` arrays and not all tests use variables, so I don't think this is necessary.

File: test/envConfig.test.ts (hunk 1 above)
Comment (lukashass) [reply test/envConfig.test.ts:105]: I will not do this for now as it makes the type of `process.env.BACKEND_URL` `string | undefined` which would complicate the test further.

Furthermore, I think this is generally less readable than just duplicating the key :thinking:

**Pull request merged**